api/utils.go

```diff
@@ -50,6 +53,40 @@ func getListOptionsForDeployment(deployment *extv1beta1.Deployment) metav1.ListO
 	}
 }
 
+func getDeploymentWithTag(deploymentName, env, branch, tag string) (*extv1beta1.Deployment, error) {
+	deploymentTemplate, err := templates.Deployment(env, branch, deploymentName)
+	if err != nil {
+		return nil, err
+	}
+	deploymentTemplate, err = deploymentTemplate.Populate(map[string]string{
+		"Tag":       tag,
+		"Namespace": "",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	deployment := new(extv1beta1.Deployment)
+	err = deploymentTemplate.Parse(deployment)
+	if err != nil {
+		return nil, err
+	}
+	return deployment, nil
+}
+
+func getImageRepoFromDeployment(deployment *extv1beta1.Deployment) (string, error) {
+	containers := deployment.Spec.Template.Spec.Containers
+	if len(containers) == 0 {
+		return "", fmt.Errorf("no containers in controller")
+	}
+	image := deployment.Spec.Template.Spec.Containers[0].Image
+	imageSplit := strings.Split(image, ":")
+	if len(imageSplit) != 2 {
```

Comment:
This should also support untagged repo references (`len(imageSplit) == 1`) so `:tag` does not need to be provided in the template.